src/cpp/server/secure_server_credentials.cc

```diff
@@ -43,6 +44,7 @@ void AuthMetadataProcessorAyncWrapper::Process(
     return;
   }
   if (w->processor_->IsBlocking()) {
+    GPR_ASSERT(w->thread_pool_);
```

Comment:
Any reason for the assertion? It looks like this was set in the constructor, correct?